.github/workflows/test.yml

```diff
@@ -0,0 +1,59 @@
+name: Test tap-postgres
+
+on: [push]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    env:
+      PIP_CONSTRAINT: .github/workflows/constraints.txt
+    strategy:
+      matrix:
+        include:
+        - {python-version: "3.10", session: "lint", os: "ubuntu-latest" }
+        - {python-version: "3.10", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.9", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.8", session: "pytest", os: "ubuntu-latest" }
+        - {python-version: "3.7", session: "pytest", os: "ubuntu-latest" }
```

Comment:
I think this is going to fail with our `cached_property` use, we could just not support `3.7`

Reply:
It passes